internal/utils/docker.go

```diff
@@ -263,6 +281,15 @@ func DockerStart(ctx context.Context, config container.Config, hostConfig contai
 	config.Labels[composeProjectLabel] = Config.ProjectId
 	// Configure container network
 	hostConfig.ExtraHosts = append(hostConfig.ExtraHosts, extraHosts...)
+	// For containers with required NOFILE limit environment ensure
+	// the container ulimit is a match
+	for _, env := range config.Env {
+		if ulimit := getUlimitFromEnvRlimit(env); ulimit != nil {
+			hostConfig.Resources.Ulimits = append(hostConfig.Resources.Ulimits, ulimit)
+			break
+		}
+	}
+
```

Comment:
I’m not entirely convinced that placing this "global docker" configuration here is the best approach, as the actual issue lies only with the realtime container. The problem I encountered is that `db start` and `start` run the realtime container in slightly different ways—one allows for the injection of `hostConfig`, while the other does not.

Since this issue could theoretically occur with any container, I thought it might make sense to implement the logic globally. However, if that feels too risky, I can limit the change to the realtime container and refactor the `db start` process to support injecting `hostConfig` with the appropriate resource configuration throughout.

Reply:
This env var is specific to realtime and is chosen somewhat arbitrarily https://github.com/supabase/realtime/blob/07a0a1500205c6698994381607064efe0d6ea270/run.sh#L6

How about removing this customisation altogether
- https://github.com/supabase/cli/blob/develop/internal/start/start.go#L762
- https://github.com/supabase/cli/blob/develop/internal/db/start/start.go#L240